src/replica/replica_stub.cpp

```diff
@@ -1331,11 +1334,20 @@ void replica_stub::on_node_query_reply(error_code err,
         }
 
         for (auto it = resp.partitions.begin(); it != resp.partitions.end(); ++it) {
+            auto meta_split_status = split_status::NOT_SPLIT;
```

Comment:
maybe you can calculate `meta_split_status` in meta server. So we don't need to calculate it in each replica server

Reply:
If an app is not splitting, `splitting_replicas` in `configuration_query_by_node_response` won't be set. The config_sync rpc will happen every 30 seconds for all replica servers, and partition split is not a frequent action, for most cases it will reduce the network cost.